app/models/driver.rb

```diff
@@ -0,0 +1,55 @@
+class Driver < ApplicationRecord
+  has_many :trips
+  validates :name, presence: true
+  validates :vin, presence: true
+
+  def total_earnings
+    earnings = 0
+    all_trips = self.trips
+
+    all_trips.each do |trip|
```

Comment:
Here, the value of `all_trips` is `self.trips`, and it's used once... Personally, I love having variables in order to give context, but I personally think that `self.trips` IS self-explanatory/self-documenting enough, that the code still reads well if we use `self.trips` directly. Consider using `self.trips` instead!

Your code might look like this, instead (deleting line 8 as well)

```suggestion
    self.trips.each do |trip|
```